extensions/hooks/_hook_copy_pdbs_to_package.py

```diff
@@ -0,0 +1,40 @@
+from conan.tools.files import copy
+import glob
+import json
+import os
+import re
+from io import StringIO
+from conans.errors import ConanException
+
+
+def post_package(conanfile):
+    if conanfile.settings.get_safe("os") != "Windows" or conanfile.settings.get_safe("compiler") != "msvc":
+        return
+    conanfile.output.info("PDBs post package hook running")
+    search_package_dll = os.path.join(conanfile.package_folder, "**/*.dll")
+    package_dll = glob.glob(search_package_dll, recursive=True)
+    if len(package_dll) == 0:
+        return
+    # Find dumpbin path
+    output = StringIO()
+    try:
+        conanfile.run(
+            r'"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -find "**\dumpbin.exe" -format json',
+            stdout=output)
+    except ConanException:
+        raise ConanException(
+            "Failed to locate dumpbin.exe which is needed to locate the PDBs and copy them to package folder.")
+    dumpbin_path = json.loads(str(output.getvalue()))[0]
+
+    for dll_path in package_dll:
+        # Use dumpbin to get the pdb path from each dll
+        dumpbin_output = StringIO()
+        conanfile.run(rf'"{dumpbin_path}" /PDBPATH {dll_path}', stdout=dumpbin_output)
+        dumpbin = str(dumpbin_output.getvalue())
```

Comment:
Usually, what paths are returned here? I just realized that for example most CCI recipes remove the pdbs in the package method, and if tyhis happens after the package, maybe the pdbs do not exist anymore for the interested recipes - I'm saying that maybe we shoukd have a way to avoid those pdb deletions in the first place?

Reply:
Can you give me an example of a recipe that does this so I can check how it works? As I understand it dumpbin gets the path of were the PDB is originally located, this is usually (always?) where the library is created which should be the build folder not the package one.

Reply:
`libxml2` is a good example.
The line you're looking for is something that starts with
`rm(self, "*.pdb", ` in the `package()` method - let me know if you need any help

Reply:
Tested this and everything is working fine as expected!

Reply:
This should be fine. Having a look at https://learn.microsoft.com/en-us/cpp/build/reference/pdbpath?view=msvc-170, it should return the path embedded in the binary, that should be the path in the build folder where the binary was originated. As this is a post_package hook, it should work even if the package() method deletes the pdbs because the hook is executed after the method 😄